kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -213,7 +220,13 @@ public void onFailure(WebSocket webSocket, Throwable t, Response response) {
 
       //If we already called onClosed() or onFailed() before, we need to abort.
```

Comment:
One more thing I noticed while debugging is that `webSocketRef` is set only in case of `onOpen`. Shall it be set just after opening the WebSocket or every time in each callback? In the testcase control gets directly to onFailure (because it's a 404). Since this socket gets closed in `closeWebSocketOnce` method .....